src/walk.rs

```diff
@@ -413,51 +407,24 @@ fn spawn_senders(
                 }
             };
 
-            if let Some(min_depth) = config.min_depth {
-                if entry.depth().map_or(true, |d| d < min_depth) {
-                    return ignore::WalkState::Continue;
-                }
-            }
-
-            // Check the name first, since it doesn't require metadata
-            let entry_path = entry.path();
-
-            let search_str: Cow<OsStr> = if config.search_full_path {
-                let path_abs_buf = filesystem::path_absolute_form(entry_path)
-                    .expect("Retrieving absolute path succeeds");
-                Cow::Owned(path_abs_buf.as_os_str().to_os_string())
-            } else {
-                match entry_path.file_name() {
-                    Some(filename) => Cow::Borrowed(filename),
-                    None => unreachable!(
-                        "Encountered file system entry without a file name. This should only \
-                         happen for paths like 'foo/bar/..' or '/' which are not supposed to \
-                         appear in a file system traversal."
-                    ),
-                }
-            };
+            let mut filters: Vec<Box<dyn Filter>> = vec![
+                Box::new(SkipRoot),
+                Box::new(MinDepth::new(config.min_depth)),
+                Box::new(RegexMatch::new(pattern.clone(), config.search_full_path)),
+                Box::new(Extensions::new(config.extensions.clone())),
+            ];
```

Comment:
Oh. You are reconstructing the filters for every single file system entry that is included in the search. This will cause 4-5 heap allocations in every single iteration.

I think we should be able to create the filters just once and share them among the threads?

Reply:
Ah yep, you're totally right. I'll fix that later and then benchmark it again.